test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/AuthorizationApplicationModelProviderTest.cs

```diff
@@ -81,6 +86,117 @@ public void CreateControllerModelAndActionModel_AllowAnonymousAttributeAddsAllow
             Assert.Single(action.Filters, f => f is AllowAnonymousFilter);
         }
 
+        [Fact]
+        public async void CreateControllerModelAndActionModel_ChangingPolicyProvider()
+        {
+            // Arrange
+            var requirements = new IAuthorizationRequirement[] {
+                new AssertionRequirement((con) => { return true; })
+            };
+            var authorizationPolicy = new AuthorizationPolicy(requirements, new string[] { "dingos" });
+            var authorizationPolicyProvider = new ChangingAuthorizationPolicyProvider();
+
+            var provider = new AuthorizationApplicationModelProvider(authorizationPolicyProvider);
+            var defaultProvider = new DefaultApplicationModelProvider(new TestOptionsManager<MvcOptions>());
+
+            var context = new ApplicationModelProviderContext(new[] { typeof(BaseController).GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+            var authorizeData = new List<IAuthorizeData> {
+                new AuthorizeAttribute("It's a Policy!")
+            };
+
+            // Act
+            provider.OnProvidersExecuting(context);
+            var firstPolicy = await AuthorizationPolicy.CombineAsync(authorizationPolicyProvider, authorizeData);
+            var secondPolicy = await AuthorizationPolicy.CombineAsync(authorizationPolicyProvider, authorizeData);
+
+            // Assert
+            Assert.Equal(1, firstPolicy.Requirements.Count);
+            Assert.Equal(2, secondPolicy.Requirements.Count);
+        }
```

Comment:
Rather than verifying Combine does different things (which AuthZ tests should already cover), could the MVC test test something more similar to exercising two requests coming in using the same filter, results in different requirements?

Maybe this test belongs in this file and just verify that NonDefaultProvider + AuthorizeFilter results in calls to the provider.

https://github.com/aspnet/Mvc/blob/91f22c8d8e102b27d278b0dee622f2fd85be36d2/test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs#L45